numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -2365,47 +2342,33 @@ array_matmul(PyObject *NPY_UNUSED(m), PyObject *args, PyObject* kwds)
     char *subscripts;
     PyArrayObject *ops[2];
 
-    npy_cache_import("numpy.core.multiarray", "matmul", &matmul);
-    if (matmul == NULL) {
-        return NULL;
-    }
-
-    errval = PyUFunc_CheckOverride((PyUFuncObject*)matmul, "__call__",
-                                   args, kwds, &override, 2, 1);
-    if (errval) {
-        return NULL;
-    }
-    else if (override) {
-        return override;
-    }
-
     if (!PyArg_ParseTupleAndKeywords(args, kwds, "OO|O:matmul", kwlist,
                                      &in1, &in2, &out)) {
         return NULL;
     }
 
-    if (out == Py_None) {
-        out = NULL;
-    }
-    if (out != NULL && !PyArray_Check(out)) {
-        PyErr_SetString(PyExc_TypeError,
-                        "'out' must be an array");
-        return NULL;
+    if (out != NULL) {
+        if (out == Py_None) {
+            out = NULL;
+        }
+        else if (!PyArray_Check(out)) {
+            PyErr_SetString(PyExc_TypeError, "'out' must be an array");
+            return NULL;
+        }
     }
 
     dtype = PyArray_DescrFromObject(in1, NULL);
     dtype = PyArray_DescrFromObject(in2, dtype);
     if (dtype == NULL) {
-        PyErr_SetString(PyExc_ValueError,
-                "Cannot find a common data type.");
+        PyErr_SetString(PyExc_ValueError, "Cannot find a common data type.");
         return NULL;
     }
     typenum = dtype->type_num;
 
     if (typenum == NPY_OBJECT) {
         /* matmul is not currently implemented for object arrays */
         PyErr_SetString(PyExc_TypeError,
-                "Object arrays are not currently supported");
```

Comment:
Not worth worrying about. I probably should have said "**_I_** break the style..." ;)